### PR TITLE
[IA-3843] (Reintroduce) When deleting RStudio or JupyterLab, redirect to Analyses tab.

### PR DESCRIPTION
**Pull request by n8landolt**
Did an improper git rebase when merging another PR and overwrote my changes for this story.

This PR reintroduces the changes requested.
Previous PR:
https://github.com/DataBiosphere/terra-ui/pull/3657/files

### CODE REVIEW

**Comment by lucymcnatt** [comment]
when I tried it on the PR site I got this error: 
image

**Comment by n8landolt** [comment]
Let me look into that @lucymcnatt

**Comment by n8landolt** [comment]
Fixed! I was attempting to pull the info from the wrong 'workspace' object.

**Review by lucymcnatt — APPROVED** [review]

File: src/pages/workspaces/workspace/analysis/AppLauncher.js [hunk 1]
```diff
@@ -254,6 +255,7 @@ const ApplicationLauncher = _.flow(
             [runtimeStatus === 'LeoReconfiguring', () => 'Cloud environment is updating, please wait.'],
             [runtimeStatus === 'Error', () => 'Error with the cloud environment, please try again.'],
             [runtimeStatus === null, () => 'Create a cloud environment to continue.'],
+            [runtimeStatus === undefined && runtime === undefined, () => Nav.goToPath(analysisTabName, { namespace, name })],
```
Comment (mlilynolting) [inline src/pages/workspaces/workspace/analysis/AppLauncher.js]: ?: This div looks like it's mostly a status display message, except in this case, where it is responsible for navigating the page. Is this responsibility here bc the div's being rendered only once everything is ready, so it's a logical spot to do the nav?

**Review by mlilynolting — APPROVED** [review]
This looks simple and workable; one question about responsibility for the nav.

File: src/pages/workspaces/workspace/analysis/AppLauncher.js (hunk 1 above)
Comment (n8landolt) [reply src/pages/workspaces/workspace/analysis/AppLauncher.js]: This ticket was a bit funky... so in short, yes.
I felt like the 'logic' took precedence over the 'role' of the conditional. I hope that makes sense.

Comment (jdcanas) [reply src/pages/workspaces/workspace/analysis/AppLauncher.js]: Hmm... I kind of agree though. Perhaps this logic should be at the top of render for this component?

I.e. !busy && runtimeStatus === undefined && runtime === undefined ? Nav.goToPath(analysisTabName, { namespace, name })

Also, does `[runtimeStatus === null, () => 'Create a cloud environment to continue.'],` even trigger with this in mind?

Comment (n8landolt) [reply src/pages/workspaces/workspace/analysis/AppLauncher.js]: Let me give that a shot!

Comment (mlilynolting) [reply src/pages/workspaces/workspace/analysis/AppLauncher.js]: the distinction between `null` and `undefined` seems like it'd be enough to trigger the different cases, but it certainly feels like a dicey architecture to have those be semantically distinct states. Probs out of the scope of this fix tho!

Comment (n8landolt) [reply src/pages/workspaces/workspace/analysis/AppLauncher.js]: @jdcanas @mlilynolting 
I don't know about runtime status being null off the top of my head. I also dislike the distinction between null and undefined.

Thoughts on the new location?

File: src/pages/workspaces/workspace/analysis/AppLauncher.js [hunk 2]
```diff
@@ -227,6 +228,8 @@ const ApplicationLauncher = _.flow(
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [googleProject, workspaceName, runtimes, bucketName])
 
+  if (!busy && runtimeStatus === undefined && runtime === undefined) Nav.goToPath(analysisTabName, { namespace, name })
```
Comment (jdcanas) [inline src/pages/workspaces/workspace/analysis/AppLauncher.js]: I dont think we need to check runtime status here, if runtime is undefined its status will trivially be so, correct?

Comment (n8landolt) [reply src/pages/workspaces/workspace/analysis/AppLauncher.js]: You're right!

**Review by jdcanas — APPROVED** [review]

**Pull request merged**